workspace/dao.py

```
@@ -32,8 +57,10 @@ def get_by_name(name):
     If the Workspace doesn't exist -> return None
     Else return the Workspace
     """
-
-    workspace = Workspace.objects(name = name)
+    try:
+        workspace = Workspace.objects(name = name)
+    except ValidationError:
+        raise EntityNotFoundException(Workspace, f"")
     if len(workspace) == 0:
         return None
```

I think this is where we'd want to raise the error instead.